src/test/java/org/candlepin/controller/PoolServiceTest.java

```diff
@@ -0,0 +1,354 @@
+        assertEquals(2, stackIds.size());
+        assertThat(stackIds, hasItems(STACK, STACK + "3"));
+
+        for (PoolUpdate update : updates) {
+            assertTrue(update.changed());
+            assertTrue(update.getQuantityChanged());
+        }
+        assertEquals((Long) 2L, stackDerivedPool1.getQuantity());
+        assertEquals((Long) 9L, stackDerivedPool2.getQuantity());
+    }
+
+    private Pool copyFromSub(Subscription sub, int id) {
+        Pool pool = TestUtil.copyFromSub(sub);
+        pool.setId("pool_" + id);
+        return pool;
+    }
+
+    private Subscription createStackedVirtSub(Owner owner, Product product, Date start, Date end) {
+        Subscription s = TestUtil.createSubscription(owner, TestUtil.createProduct());
+        s.setStartDate(start);
+        s.setEndDate(end);
+        s.setProduct(product.toDTO());
+        s.setContractNumber(Integer.toString(TestUtil.randomInt()));
+        s.setOrderNumber(Integer.toString(TestUtil.randomInt()));
+        s.setAccountNumber(Integer.toString(TestUtil.randomInt()));
+        return s;
+    }
+
+    @Test
+    public void bulkUpdateDoesNotDeletesPoolsWithoutStackingEntitlements() {
+        Owner owner = TestUtil.createOwner();
+        Consumer consumer = TestUtil.createConsumer(owner);
+        Set<Consumer> consumers = Collections.singleton(consumer);
+        List<Pool> pools = createPools(owner);
+        List<Entitlement> stackingEntitlements = createEntitlements(owner, consumer, pools);
+        when(entitlementCurator.findByStackIds(isNull(), anyCollection()))
+            .thenReturn(stackingEntitlements);
+
+        poolService.bulkUpdatePoolsFromStack(consumers, pools, new ArrayList<>(), false);
+
+        verifyNoInteractions(poolCurator);
+    }
+
+    @Test
+    public void bulkUpdateDeletesPoolsWithoutStackingEntitlements() {
+        Owner owner = TestUtil.createOwner();
+        Consumer consumer = TestUtil.createConsumer(owner);
+        Set<Consumer> consumers = Collections.singleton(consumer);
+        List<Pool> pools = createPools(owner);
+        List<Entitlement> stackingEntitlements = createEntitlements(owner, consumer, pools);
+        pools.add(createPool(owner));
+        when(entitlementCurator.findByStackIds(isNull(), anyCollection()))
+            .thenReturn(stackingEntitlements);
+        when(poolCurator.lockAndLoad(anyCollection())).thenReturn(pools);
+
+        poolService.bulkUpdatePoolsFromStack(consumers, pools, new ArrayList<>(), true);
+
+        verify(poolCurator).batchDelete(anyCollection(), anyCollection());
+    }
+
+    @Test
+    public void testNullArgumentsDontBreakStuff() {
+        Owner owner = TestUtil.createOwner();
+
+        poolService.getBySubscriptionIds(owner.getId(), null);
+        poolService.getBySubscriptionIds(owner.getId(), new ArrayList<>());
+        poolService.getBySubscriptionId(owner, null);
+    }
+
+    @Test
+    public void testgetBySubscriptionIds() {
```

Comment:
Camel case: `testgetBySubscriptionIds` to `testGetBySubscriptionIds`

Reply:
Done.